cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1362,7 +1362,10 @@ private boolean hasRebalanceAnnotation(KafkaRebalance kafkaRebalance) {
 
     private boolean shouldRenewRebalance(KafkaRebalance kafkaRebalance, KafkaRebalanceAnnotation rebalanceAnnotation) {
         // check if refresh annotation applied or request are made when CC is not active
-        if (rebalanceAnnotation == KafkaRebalanceAnnotation.refresh || kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> "ConnectException".equals(condition.getReason()))) {
+        // We check for `CruiseControlRetriableConnectException` if request are made when CC was not active since not handling this case will move the KafkaRebalance to `NotReady`
+        // and it will stay in `NotReady` state until we apply `refresh`
+        if (rebalanceAnnotation == KafkaRebalanceAnnotation.refresh
+                || kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> "CruiseControlRetriableConnectException".equals(condition.getReason()))) {
```

Comment:
should it be "CruiseControlRetriableConnectionException" now? Even in the comment above?